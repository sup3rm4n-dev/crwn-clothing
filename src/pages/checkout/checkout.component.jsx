import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import CheckOutItem from '../../components/checkout-item/checkout-item.component';
import StripeCheckoutButton from '../../components/stripe-button/stripe-button.component';
import {
  selectCartItems,
  selectCartTotal,
} from '../../redux/cart/cart.selectors';
import {
  CheckoutPageContainer,
  CheckOutHeaderContainer,
  HeaderBlock,
  TotalContainer,
  WarningContainer,
} from './checkout.styles';

const CheckOutPage = ({ cartItems, total }) => (
  <CheckoutPageContainer>
    <CheckOutHeaderContainer>
      <HeaderBlock>
        <span>PRODUCT</span>
      </HeaderBlock>
      <HeaderBlock>
        <span>DESCRIPTION</span>
      </HeaderBlock>
      <HeaderBlock>
        <span>QUANTITY</span>
      </HeaderBlock>
      <HeaderBlock>
        <span>PRICE</span>
      </HeaderBlock>
      <HeaderBlock>
        <span>REMOVE</span>
      </HeaderBlock>
    </CheckOutHeaderContainer>
    {cartItems.map(cartItem => (
      <CheckOutItem key={cartItem.id} cartItem={cartItem} />
    ))}
    <TotalContainer>
      <span>TOTAL ${total}</span>
    </TotalContainer>
    <WarningContainer>
      *Please use the following for test credit card payments*
      <br />
      4242 4242 4242 4242 - Exp: 01/20 - CVV: 123
    </WarningContainer>
    <StripeCheckoutButton price={total} />
  </CheckoutPageContainer>
);

const mapStateToProps = createStructuredSelector({
  cartItems: selectCartItems,
  total: selectCartTotal,
});

export default connect(mapStateToProps)(CheckOutPage);
