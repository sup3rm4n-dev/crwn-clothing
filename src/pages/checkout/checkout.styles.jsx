import styled from 'styled-components/macro';

import { media } from '../../helpers';

export const CheckoutPageContainer = styled.div`
  width: 60%;
  min-height: 80vh;
  margin: 5rem auto 0;
  display: block;
  button {
    margin-left: auto;
    margin-top: 5rem;
  }
  ${media.tablet`width: 90%;`}
`;

export const CheckOutTable = styled.table`
  display: table;
  text-align: center;
  width: 100%;
  border-spacing: 0;
  border-collapse: collapse;
`;

export const TableHeader = styled.thead`
  border-bottom: 1px solid darkgrey;
  font-size: 2rem;
  text-transform: uppercase;
  padding: 1rem 0;
  width: 100%;
`;

export const TableRow = styled.tr`
  padding: 0;
  margin: 0 auto;
  border-bottom: 1px solid darkgrey;
`;

export const TableHead = styled.th`
  text-align: center;
  font-weight: 300;
  font-size: 2.4rem;
  &:first-child {
    text-align: left;
    width: 25%;
    padding-left: 1.5rem;
  }
  &:last-child {
    text-align: right;
    padding-right: 1.5rem;
  }
  ${media.tablet`
    font-size: 2.2rem;
    &:first-child {
      padding-left: 0;
    }
    &:last-child {
      padding-right: 0;
    }
  `}
  ${media.phablet`font-size: 2rem;`}
`;

export const CheckOutHeaderContainer = styled.div`
  width: 100%;
  padding: 1rem 0;
  display: flex;
  justify-content: space-between;
  border-bottom: 1px solid darkgrey;
  font-size: 2rem;
`;

export const HeaderBlock = styled.div`
  display: inline-block;
  text-transform: uppercase;
`;
