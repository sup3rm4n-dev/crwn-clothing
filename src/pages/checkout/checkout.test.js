import React from 'react';
import { shallow } from 'enzyme';
import { CheckOutPage } from './checkout.component';

describe('CheckOutPage component', () => {
  let wrapper;

  beforeEach(() => {
    const mockProps = {
      cartItems: [],
      total: 100,
    };
    wrapper = shallow(<CheckOutPage {...mockProps} />);
  });

  it('should render CheckOutPage component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
