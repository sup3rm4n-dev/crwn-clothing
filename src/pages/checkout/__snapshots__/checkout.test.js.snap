// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckOutPage component should render CheckOutPage component 1`] = `ShallowWrapper {}`;
